src/Database/DBStructure.php

```diff
@@ -425,6 +425,38 @@ public static function install(string $basePath)
 		return self::update($basePath, false, true, true);
 	}
 
+	/**
+	 * Explicit updates the Lock table
+	 * This is necessary in case of self-updating the lock table
+	 *
+	 * @param string  $basePath
+	 * @param boolean $verbose
+	 * @param boolean $action
+	 * @param boolean $install
+	 *
+	 * @throws Exception
+	 */
+	public static function updateLockTable(string $basePath, bool $verbose, bool $action, bool $install)
```

Comment:
@annando I'm not 100% sure if this is an good idea ...

This function will be called **before** any locks are used during the update and should update/create the `host` and `locks` table beforehand (without any locking mechanism ...)

At my local node, I was able to upgrade from version 1290 (`friendica_test_data.sql`) to this version..

Reply:
This sounds a little bit like the existing function `checkInitialValues`

Reply:
But it's a structure upgrade, not an insert of values.. But yeah, it's similar I think

Reply:
But the main question: does it work? :D .. For simple updates it will.. But are there circumstances where a structure update with an own dba lock before the main update lock could lead to problems? 

I think I could add another check if there's a update lock and stop the update ..

Reply:
I'm not even sure if this does work at all.